### DISCUSSION TITLE
OPC Ua Timestamp

### DISCUSSION THREAD

**Discussion by Praveenstein** [apache/streampipes#886] [Q&A] [answered]
While using the opc ua adapter to query some node, how do i get the server timestamp also along with the data?? it's currently only reading the value as shown below.
![e1](https://user-images.githubusercontent.com/45847217/209311243-8d004729-7bc7-4e04-aff9-0883a11feeda.png)
![e2](https://user-images.githubusercontent.com/45847217/209311251-f487d3c0-fbc4-4af6-9a84-6ddb03bb08ee.png)

**Comment by bossenti** [comment]
You can click the 'add timestamp' button this extends your message schema with a timestamp and allows you to proceed here.

**Reply by tenthe** [reply]
You can do it as @bossenti described. Currently, the timestamp is added within the adapter that is created. 
As a user would you also like the option to use the timestamp of the OPC UA server?

**Reply by Praveenstein** [reply]
I did try the approach of adding time by the adapter, but It would be better if we could get the time from the actual server itself

**Reply by tenthe** [reply, marked as answer]
Ok, I hope this solution works for you for now.
I have created an issue to provide the option to use the OPC UA source time as timestamp (#889).

**Reply by bossenti** [reply]
thanks @tenthe 
@Praveenstein if you have anything to add, feel free to comment on the linked issue.
We really appreciate your thoughts 🙏🏼
